mage_ai/server/api/downloads.py

```diff
@@ -62,3 +66,39 @@ async def get(self, pipeline_uuid, block_uuid, **kwargs):
                     await self.flush()
                     # Sleep for a nanosecond so other handlers can run and avoid blocking
                     await gen.sleep(0.000000001)
+
+
+class ApiResourceDownloadHandler(BaseHandler):
+
+    def get(self, token):
+        try:
+            decoded_payload = jwt.decode(token, JWT_SECRET, algorithms=["HS256"])
+
+            file_path = decoded_payload['file_path']
+            file_name = file_path.split('/')[-1]
+            is_temporary = file_path.startswith(TEMPORARY_DOWNLOAD_LOCATION)
+            with open(file_path, "rb") as f:
```

Comment:
I feel there's a security vulnerability.

User can manually generate the token if they know the JWT_SECRET (the default value is public) and then they can download any file in the instance.

Reply:
While I believe having a default value for something as critical as a JWT secret is a security vulnerability in and of itself, we've taken measures to address this concern. At startup, we now replace the default secret with a randomly generated string value. This step effectively eliminates the risk of token forgery. However, it's important to note that this change has consequences: in the event of an environment reset, all active user sessions will become invalid.
Since the two token types are different in both scope and lifespan, we've decided to separate their secrets. The OAuth tokens will continue to use the original JWT_SECRET, while the download tokens will be generated using a new secret, JWT_DOWNLOAD_SECRET, which would get randomly generated if a specific value is not provided at startup. We're confident about this solution as it strikes a balance between security and user experience, hope you guys like it as well.